compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1218,9 +1218,13 @@ impl<'tcx> InferCtxt<'tcx> {
         );
 
         if self.tcx.sess.err_count() > self.err_count_on_creation {
```

Comment:
Oh dear, this is causally flawed... nowadays we can have multiple `InferCtxt`s alive at the same time (for different query instances on the stack, that depend on eachother).

The `delay_span_bug` should never cause an issue here but... `err_count_on_creation` could cause _incremental unsoundness_ due to errors _produced elsewhere_ in the system.

At the very least this kind of thing needs far more clever logic to save/restore counts/deltas when entering/exiting `InferCtxt`s and I'm not even sure it's worth it... (cc @jackh726 @nikomatsakis).

Reply:
i thought we don't cache anything on disc if there are any errors during this compilation session? :thinking: